myhpom/tests/test_dashboard_view.py

```diff
@@ -62,3 +64,45 @@ def test_user_is_organ_donor(self):
         response = self.client.get(self.url)
         self.assertIn('I am an organ donor', response.content)
         self.assertNotIn('Learn how you can be a donor.', response.content)
+
+    def test_user_verification(self):
+        """
+        + Unverified user
+            + sees a warning message on the dashboard (alert-warning)
+            + has "Upload your File" button disabled
+        + Verified user
+            + sees no warning message on the dashboard
+            + has "Upload your File" button enabled
+        """
+        user = UserFactory()
+        user.set_password('password')
+        user.save()
+        self.assertTrue(self.client.login(username=user.email, password='password'))
+
+        # unverified user
+        user.userdetails.reset_verification()
+        user.userdetails.save()
+        self.assertIsNotNone(user.userdetails.verification_code)
+        self.assertIsNone(user.userdetails.verification_completed)
+        response = self.client.get(self.url)
+        messages = [msg for msg in get_messages(response.wsgi_request) if msg.level == WARNING]
+        self.assertGreater(len(messages), 0)
+        matchdata = re.search(
```

Comment:
I think this way of doing it is fine for this PR's purposes, but in the future we might should consider actually parsing the generated HTML and inspecting the resulting element tree—it's a bit more reliable than string searches.

Reply:
That's a good idea.